Control/Monad/Logic.hs

```diff
@@ -120,6 +120,32 @@ observeManyT n m
 -------------------------------------------------------------------------
 -- | Runs a 'LogicT' computation with the specified initial success and
 -- failure continuations.
+--
+-- The second argument takes one result of the 'LogicT' computation
+-- and the monad to run for any subsequent matches.
+--
+-- The third argument is called when the 'LogicT' cannot produce any
+-- more results.
+--
+-- For example:
+--
+-- >>> yieldWords l = if null l then mzero else return (head l) `mplus` yieldWords (tail l)
```

Comment:
I'm hesitant about this definition. What if we remove it and just use ``return "foo" `mplus` return "bar"`` instead of `yieldWords ["foo", "bar"]` and `mzero` instead of `yieldWords []` below? It would make examples more self-sufficient.

Reply:
I'm happy to add an mplus/mzero only example, but I think it helps users to see different forms that those can be used from.  I've tried to use some variation to help provide examples that might suggest some uses that people would find illuminating.  [To that end, I'm working on some documentation updates in the Class file, along with some unit test additions.]

For various reasons, I changed it away from a pattern match, but perhaps returning to that would be clearer:
```
yieldWords [] = mzero
yieldWords (a:as) = return a `mplus` yieldWords as
```

What do you think of that form?

Reply:
I agree that it makes sense to show different forms?

How about `yieldWords = foldr (mplus . return) mzero` or `yieldWords = msum . map return`? Is it too cryptic?

Reply:
I like the foldr form: I think it is both a good alternative example and is fairly intuitive for folks.